api/scpca_portal/loader.py

```diff
@@ -181,6 +181,17 @@ def _create_computed_file_callback(future, *, update_s3: bool, clean_up_output_d
     connection.close()
 
 
+def process_dataset(
+    dataset,
+    update_s3: bool = True,
+) -> None:
+    if old_dataset_file := dataset.computed_file:
+        old_dataset_file.purge(update_s3)
+
+    dataset_file = ComputedFile.get_dataset_file(dataset)
+    _create_computed_file(dataset_file, update_s3, clean_up_output_data=False)
+
```

Comment:
I wouldn't put this here since it isn't loader related. I think you also don't want to call `_create_computed_file`. Instead you should just pull in the part of logic you want. That really is just uploading the output file and the clean up.